src/lib/utils/data_src.h

```diff
@@ -121,8 +122,15 @@ class BOTAN_PUBLIC_API(2,0) DataSource_Memory final : public DataSource
       * Construct a memory source that reads from a secure_vector
       * @param in the MemoryRegion to read from
       */
-      explicit DataSource_Memory(const secure_vector<uint8_t>& in) :
-         m_source(in), m_offset(0) {}
+      explicit DataSource_Memory(secure_vector<uint8_t> in) :
+         m_source(std::move(in)), m_offset(0) {}
```

Comment:
`DataSource_Memory` is creating an internal copy of the data. With this patch, users can avoid that if necessary by moving the data buffer in.